geode-management/src/main/java/org/apache/geode/management/api/Links.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.management.api;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import org.apache.commons.lang3.StringUtils;
+
+public class Links {
```

Comment:
Add javadocs for this class. It is an external api. I find the class confusing. I expected it to just be a map but instead it has self, list, and others. It is also not clear how the constants on this class are related to links.
It would be best for this to be an interface so that it could be read only.
We really don't intend for users to be calling any of these settors or constructors.
If we can't make it an interface then get rid of any of the settors that we can. For example get rid of addLink. We don't need it for our implementation. We can instead just call getOthers().put(key, url)

Reply:
moved this to internal package. User doesn't need to use this class.